src/orange/OrangeParams.cc

```diff
@@ -125,22 +141,30 @@ OrangeParams::OrangeParams(OrangeInput input)
                       "stack is limited to a depth of "
                    << detail::LogicStack::max_stack_depth());
 
-    // TODO: update this to work over multiple universe levels
+    std::vector<Label> surface_labels;
+    std::vector<Label> volume_labels;
+
+    for (const UnitInput& u : input.units)
     {
         // Capture metadata
-        UnitInput& u = input.units.front();
-        surf_labels_ = LabelIdMultiMap<SurfaceId>{std::move(u.surfaces.labels)};
-        std::vector<Label> volume_labels;
-        volume_labels.resize(u.volumes.size());
-        for (auto i : range(u.volumes.size()))
-        {
-            volume_labels[i] = std::move(u.volumes[i].label);
-        }
-        vol_labels_ = LabelIdMultiMap<VolumeId>{std::move(volume_labels)};
-        bbox_       = u.bbox;
+
+        std::copy(u.surfaces.labels.begin(),
+                  u.surfaces.labels.end(),
+                  std::back_inserter(surface_labels));
```

Comment:
Nice use of algorithm! Better yet, though:
```suggestion
        surface_labels.insert(surface_labels.insert .end(),
                u.surfaces.labels.begin(),
                u.surfaces.labels.end());
```